### PR TITLE
MMDLoader update

### PR DESCRIPTION
**Pull request by takahirox**
I updated MMDLoader and enabled it to handle more various models.

For example, these two models weren't displayed correctly before but they're fine now.

https://twitter.com/superhoge/status/674102177867022336
https://twitter.com/superhoge/status/674840872404451328

### CODE REVIEW

File: examples/js/loaders/MMDLoader.js [hunk 1]
```diff
@@ -2179,9 +2198,135 @@ THREE.MMDLoader.prototype.createMesh = function ( model, texturePath, onProgress
...
+							c.height = image.height;
+
+							var ctx = c.getContext( '2d' );
+							ctx.drawImage( image, 0, 0 );
+
+							return ctx.getImageData( 0, 0, c.width, c.height );
+
+						};
+
+						function detectTextureTransparency ( image, uvs ) {
```
Comment (mrdoob) [inline examples/js/loaders/MMDLoader.js]: Isn't it simpler to parse all the pixels in the imageData?

```javascript
var data = ctx.getImageData( 0, 0, c.width, c.height );
for ( var i = 3, l = data.length; i < l; i += 4 ) {
    if ( data[ i ] > threshold ) return true;
}
return false;
```

Comment (takahirox) [inline examples/js/loaders/MMDLoader.js]: MMD decides whether a certain material is transparent or not by
checking the alpha value actually used.
So I need to judge with uv.

For examples, assume
Material A and Material B uses the same Texture image partially including alpha < 1.0.
If Material A uses the part of the image including alpha < 1.0 while
Material B uses the part not including alpha < 1.0, 
then Material A will be transparent but Material B won't be.

Comment (mrdoob) [inline examples/js/loaders/MMDLoader.js]: Oh, I see... Would setting `transparent` for both materials create artefacts?

Comment (takahirox) [inline examples/js/loaders/MMDLoader.js]: Ah, sorry, English question. What's "artefacts" in this case?

Comment (mrdoob) [inline examples/js/loaders/MMDLoader.js]: I mean that, if you instead just checked the pixels of the whole texture and after finding a transparent pixel then you would make both Material A and Material B transparent, would that render incorrectly?

Comment (takahirox) [inline examples/js/loaders/MMDLoader.js]: I tried it but some models were displayed incorrectly.
Indeed, this model was one of them.

https://twitter.com/superhoge/status/674102177867022336

This detection seems necessary and some other MMD tools(i.e. MMD Plugin for Unity)
have similar detection logic.

(TBH, I didn't think this's necessary but I ran into models which are rendered incorrectly
 without the detection, so I decided to implement)

Comment (mrdoob) [inline examples/js/loaders/MMDLoader.js]: I see I see. Maybe we can add a comment though? Something like...

```
// Check if this part of the texture requires transparency
```

Comment (takahirox) [inline examples/js/loaders/MMDLoader.js]: done :D

Comment (mrdoob) [inline examples/js/loaders/MMDLoader.js]: sweet!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**